class-udm-updater.php

```diff
@@ -97,6 +102,28 @@ public function __construct($mothership, $muid, $relative_plugin_file, $options
 		}
 	}
 
+	/**
+	 * Potentially disconnect a cloned site by unsetting the 'email' option and resetting PUC update cache and at the same time adding 'site_host_path' as a plugin option for a cloned site checking purpose in future
+	 */
+	public function potentially_disconnect_cloned_site() {
+
+		$udm_options = $this->get_option($this->option_name);
+		$site_url = parse_url(network_site_url());
+		$site_host_path = isset($site_url['host']) ? $site_url['host'] : '';
+		$site_host_path .= isset($site_url['path']) ? $site_url['path'] : '';
+		/**
+		 * If it's a cloned site, other plugins that use the same site ID may have still connected. Just unset the email and clear the PUC update cache so that when the users reconnect they get the warning message about the duplicate site ID issue
+		 * any releases of UDM that doesn't have the `site_host_path` option must first have their `email` option unset and PUC update cache reset, including any plugin that was cloned from a site where the plugin is connected but it has a different URL site compared to the site_host_path option
+		 */
+		if (empty($udm_options['site_host_path']) || (!empty($udm_options['email']) && strtolower($udm_options['site_host_path']) !== strtolower($site_host_path))) {
```

Comment:
@andtown Since `site_host_path` doesn't exist in the saved option on any existing installs, doesn't that means that this code will disconnect all existing installs that are using this class?

Reply:
@DavidAnderson684 Yes, correct. It's going to disconnect all existing installs. What I'm currently aiming by doing it is that to make sure that all cloned sites get disconnected first so that when accessing plugins or update page, users become aware of it and soon they reenter their credentials they will find that their site was previously using a licence that was also used by another WordPress install with a different URL. However, since currently no mechanism to detect which sites are duplicated, all existing installs will be disconnected no matter what their state is. I can't see any other way, the only options are not to disconnect all or disconnect all. Any thoughts on this?

Reply:
@andtown Let's just not worry about the existing sites. Don't disconnect them all.
(Please mention this sort of thing explicitly in future - it wasn't mention in the card and would have had a widespread effect).

Reply:
@DavidAnderson684 Okay, well noted. I've updated the code.